### ISSUE TITLE
Deleting connected output crashed rafcon

### ISSUE THREAD

**Issue by franzlst** [DLR-RM/RAFCON#834]
When deleting an output from a hierarchy statemachine which is connected to internal states rafcon experiences undefined bahaviour. 

Furthermore you cannot open the statemachine afterwards because of the following error:

```
Could not open library: Transition invalid within state mission_control (id XXPUOJ): to_outcome is not existing
```

Here some error output:

```
Traceback (most recent call last):
  File "/internal_path/common/packages/rafcon/0.6.0/lib/python2.7/rafcon/mvc/controllers/graphical_editor.py", line 425, in _on_mouse_release
    new_selection = self._find_selection(event.x, event.y)
  File "/internal_path/common/packages/rafcon/0.6.0/lib/python2.7/rafcon/mvc/controllers/graphical_editor.py", line 1855, in _find_selection
    self.view.editor.prepare_selection(pos_x, pos_y, width, height)
  File "/internal_path/common/packages/rafcon/0.6.0/lib/python2.7/rafcon/mvc/views/graphical_editor.py", line 729, in prepare_selection
    glSelectBuffer(self.name_counter * 6)
  File "/internal_path/moro/packages/gtk_etc_python/0.0.5/sled11-x86-gcc4.x/lib/python2.7/site-packages/PyOpenGL-3.1.0-py2.7.egg/OpenGL/GL/pointers.py", line 225, in glSelectBuffer
    _simple.glSelectBuffer( size, buffer )
  File "/internal_path/moro/packages/gtk_etc_python/0.0.5/sled11-x86-gcc4.x/lib/python2.7/site-packages/PyOpenGL-3.1.0-py2.7.egg/OpenGL/error.py", line 232, in glCheckError
    baseOperation = baseOperation,
OpenGL.error.GLError: GLError(
    err = 1282,
    description = 'invalid operation',
    baseOperation = glSelectBuffer,
    cArguments = (
        12,
        array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], dtype=uint32),
    )
)
```

Originally created by @peter-lehner  at 2016-01-20 13:51:27+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
I cannot reproduce your problem. I know that there had been this issue, but it should have been fixed. You are using version 0.6.0, right?

Maybe you can send me an example state machine, where this problem occurs, or you call me and show it to me live :smile: 


Originally created by @franzlst  (Franz.Steinmetz@dlr.de) at 2016-01-20 14:18:09+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
I don't have had a crash, but similar behavior when I was removing outcome with internal connected transitions. In my case I used the OpenGL version of the StateMachineEditor. It was caused by an KeyError Exception while drawing the transition which was removed with the outcome. The drawing function could not access some Variable in the state_m.temp-dictionary (combined meta data). It was only reproducable by using the ChangeHistory and I firstable catched this Exception and ignore it at the moment.
-> This Issue has been solved by commit e2a5e5b33666a917380b428ece9ee133ab170396


Originally created by @Rbelder  at 2016-01-22 18:40:40+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
I think the issue mentioned by @peter-lehner is not related to the issue solved by commit e2a5e5b. However, we will need @peter-lehner feedback for this.


Originally created by @franzlst  (Franz.Steinmetz@dlr.de) at 2016-01-26 08:16:18+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
Does not seem to exist any more.


Originally created by @sebastian-brunner  (Sebastian.Brunner@dlr.de) at 2016-02-22 14:31:25+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
Steps to reproduce:
1. Open /internal_path/peter-lehner/public/rafcon/outcome_fail
2. Select top level state machine outcome_fail
3. In state editor click:
   1. Outcomes tab
   2. Select 'success' outcome
   3. Click remove
   4. tada


Originally created by @peter-lehner  at 2016-02-23 14:32:03+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
@peter-lehner I tried to reproduce the failure on a 32-bit machine with the develop-branch on commit 8806a6e89 and I could not reproduce it. The failure. Are you may on the master or on which revision/commit your are working?


Originally created by @Rbelder  at 2016-02-23 16:25:31+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
I was using the latest released version from RMPM (0.6.0).

I tried it with the develop branch and I couldn't reproduce it either. So I guess its fixed but not released jet.


Originally created by @peter-lehner  at 2016-02-23 16:46:53+00:00 (moved from RMC internal repository)
